src/Middleware.php

```diff
@@ -186,7 +186,24 @@ public static function retry(
             return new RetryMiddleware($decider, $delay, $handler, $stats);
         };
     }
-
+    /**
+    * Middleware wrapper function that adds an invocation id header to requests.
+    * This is a uniquely generated UUID to identify initial and subsequent retries as part of a complete request lifecycle.
+    *
+    * @return callable
+    */
+    public static function invocationId()
+    {
+        return function (callable $handler) {
+            return function (
+                CommandInterface $command,
+                RequestInterface $request
+            ) use ($handler){
+                $request = $request->withHeader('aws-sdk-invocation-id', md5(uniqid()));
```

Comment:
This will need to be more unique than just `uniqid`. I think using `gethostname` was a good choice but would prefer to see the hostname normalized. How about `uniqid(md5(gethostname(), true)`?